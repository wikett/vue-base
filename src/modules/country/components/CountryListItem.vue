<script setup lang="ts">
import type { Country } from '@/modules/country/services/types'
import { GlobeEuropeAfricaIcon, UserGroupIcon } from '@heroicons/vue/20/solid'
const props = defineProps<{
  country: Country
}>()
</script>
<template>
  <li
    :key="props.country.name.official"
    class="col-span-1 my-1 flex flex-col divide-y divide-gray-200 rounded-lg bg-transparent text-center shadow shadow-cyan-700 transition-opacity duration-100 ease-out"
  >
    <div class="flex flex-1 flex-col px-8 pt-8">
      <img
        class="mx-auto h-32 w-full flex-shrink-0 rounded-md object-contain"
        :src="props.country.flags.svg"
        :alt="props.country.flags.alt"
      />
      <h3 class="mt-6 text-sm font-semibold text-secondary-800">
        {{ props.country.name.official }}
      </h3>
      <dl class="mt-1 flex flex-grow flex-col justify-between">
        <dd v-if="props.country.capital" class="mt-3">
          <span
            class="inline-flex items-center rounded-full bg-green-50 px-2 py-1 text-xs font-medium text-green-700 ring-1 ring-inset ring-green-600/20"
            >{{ props.country.capital[0] }}</span
          >
        </dd>
      </dl>
    </div>
    <div>
      <div class="-mt-px flex divide-x divide-gray-200">
        <div class="flex w-0 flex-1">
          <div
            class="relative inline-flex w-0 flex-1 items-center justify-center gap-x-3 rounded-br-lg border border-transparent py-4 text-sm font-semibold text-gray-900"
          >
            <div class="flex flex-col items-center">
              <GlobeEuropeAfricaIcon class="h-5 w-5 text-gray-400" aria-hidden="true" />
              <p class="text-sm font-semibold">{{ props.country.continents[0] }}</p>
            </div>
          </div>
        </div>
        <div class="-ml-px flex w-0 flex-1">
          <div
            class="relative inline-flex w-0 flex-1 items-center justify-center gap-x-3 rounded-br-lg border border-transparent py-4 text-sm font-semibold text-gray-900"
          >
            <div class="flex flex-col items-center">
              <UserGroupIcon class="h-5 w-5 text-gray-400" aria-hidden="true" />
              <p>{{ props.country.population.toLocaleString() }}</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </li>
</template>
