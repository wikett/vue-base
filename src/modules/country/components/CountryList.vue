<script setup lang="ts">
import type { Country } from '@/modules/country/services/types'
import CountryListItem from '@/modules/country/components/CountryListItem.vue'

const props = defineProps<{
  countries: Country[]
}>()
</script>
<template>
  <div class="mx-auto mt-12 max-w-7xl overflow-hidden sm:px-6 lg:px-8">
    <TransitionGroup
      tag="ul"
      name="list"
      class="grid grid-cols-1 gap-6 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-5"
      role="list"
    >
      <CountryListItem
        v-for="country in props.countries"
        :key="country.name.official"
        :country="country"
      />
    </TransitionGroup>
  </div>
</template>
<style scoped>
.list-enter-active,
.list-leave-active {
  transition: all 0.5s ease;
}
.list-enter-from,
.list-leave-to {
  opacity: 0;
}
</style>
1
