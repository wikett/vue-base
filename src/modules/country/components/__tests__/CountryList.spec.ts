import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import CountryList from '@/modules/country/components/CountryList.vue'
import CountryListItem from '@/modules/country/components/CountryListItem.vue'
import { countryOne, countryTwo } from '@/modules/country/components/__tests__/fakeCountry'

describe('CountryList', () => {
  it('should render correctly', () => {
    const wrapper = mount(CountryList)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have a CountryListItems', () => {
    const testContries = []
    testContries.push(countryOne)
    const wrapper = mount(CountryList, {
      props: {
        countries: testContries
      }
    })

    expect(wrapper.findComponent(CountryListItem).exists()).toBe(true)
  })

  it('should have more than 1 CountryListItems', () => {
    const testContries = []
    testContries.push(countryOne, countryTwo)
    const wrapper = mount(CountryList, {
      props: {
        countries: testContries
      }
    })

    expect(wrapper.findAllComponents(CountryListItem).length).greaterThan(1)
  })
})
