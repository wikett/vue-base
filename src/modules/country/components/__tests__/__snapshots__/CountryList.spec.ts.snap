// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CountryList > should render correctly 1`] = `
"<div data-v-d284c788="" class="mx-auto mt-12 max-w-7xl overflow-hidden sm:px-6 lg:px-8">
  <transition-group-stub data-v-d284c788="" tag="ul" name="list" appear="false" persisted="false" css="true" class="grid grid-cols-1 gap-6 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-5" role="list"></transition-group-stub>
</div>"
`;
